From ca463a294424f1e8cecea5818b8dc9022e4f3028 Mon Sep 17 00:00:00 2001
From: Alexander Narsudinov <a.narsudinov@gmail.com>
Date: Sun, 7 Nov 2021 15:03:20 +0300
Subject: [PATCH] Fix incorrect power profile support validation.

Before this patch power profile support validation used wrong flag from PlatformProfileFunctions struct.
---
 asusctl/src/main.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/asusctl/src/main.rs b/asusctl/src/main.rs
index 8eaf198..d6cdc26 100644
--- a/asusctl/src/main.rs
+++ b/asusctl/src/main.rs
@@ -415,7 +415,7 @@ fn handle_profile(
     supported: &PlatformProfileFunctions,
     cmd: &ProfileCommand,
 ) -> Result<(), Box<dyn std::error::Error>> {
-    if !supported.fan_curves {
+    if !supported.platform_profile {
         println!("Profiles not supported by either this kernel or by the laptop.");
         return Err(ProfileError::NotSupported.into());
     }
-- 
GitLab

